samples/Calling/src/app/views/CallScreen.tsx

```diff
@@ -37,7 +37,12 @@ export const CallScreen = (props: CallScreenProps): JSX.Element => {
 
   // Whenever the sample is changed from desktop -> mobile using the emulator, make sure we update the formFactor.
   useEffect(() => {
-    const updateIsMobile = (): void => setIsMobileSession(detectMobileSession());
+    const updateIsMobile = (): void => {
```

Comment:
I am curious why we want to call this in useEffect. useEffect is called after things are rendered.

Why cant we just use mobileDetect once at the very beginning and pass that prop through the components?

Reply:
So, this useEffect will be called only once as soon as this composite loads since it has an empty dependency array.

Reply:
I guess my question is why do we even need to call useEffect to run mobileDetect

Reply:
The `useEffect` is needed to register window events, but only once.